spring-infrastructure/src/main/java/org/spring/training/boot/infrastructure/external/bpp/TrimmedAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,55 @@
+package org.spring.training.boot.infrastructure.external.bpp;
+
+import org.apache.commons.lang3.StringUtils;
+import org.spring.training.boot.infrastructure.external.annotation.Trimmed;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.cglib.proxy.Enhancer;
+import org.springframework.cglib.proxy.MethodInterceptor;
+
+public class TrimmedAnnotationBeanPostProcessor implements BeanPostProcessor {
+
+    @Override
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        Class<?> beanClass = bean.getClass();
+        Trimmed trimmedAnnotation = beanClass.getAnnotation(Trimmed.class);
+        if (trimmedAnnotation != null) {
+            return enhanceBean(beanClass, trimmedAnnotation);
```

Comment:
It looks like you just throw away bean with all configuration that was previously applied by other bean post processors if the class is annotated with Trimmed annotation.